Don’t require `NIX_PATH` entries to be valid paths

thufschmitt opened the pull request:
It’s totally valid to have entries in `NIX_PATH` that aren’t valid paths
(they can even be arbitrary urls or `channel:<channel-name>`).

Fix #5598 and #5980

/cc @Mic92 @shlevy

shlevy reviewed (approved):
Awesome, thanks!

shlevy commented:
@thufschmitt FYI typo in your commit message referring to my issue, I fixed it in the PR description already.

thufschmitt commented:
> @thufschmitt FYI typo in your commit message referring to my issue, I fixed it in the PR description already.

Oh thanks. Fixed in the commit message too

Mic92 commented:
Thanks. a lot this fixes my problem!

shlevy commented:
@thufschmitt @edolstra Not sure why the backport broke, but should this be on the maint branch? Would be great to get a hotfix out.
